src/wazuh_modules/vulnerability_scanner/tests/unit/factoryOrchestrator_test.cpp

```diff
@@ -10,84 +10,539 @@
  */
 
 #include "factoryOrchestrator_test.hpp"
-#include "MockCleanAgentInventory.hpp"
-#include "MockDetailsAugmentation.hpp"
-#include "MockFetchAllGlobalDb.hpp"
-#include "MockInventorySync.hpp"
-#include "MockOsScanner.hpp"
-#include "MockPackageScanner.hpp"
-#include "MockQueryAllPkgsAgt.hpp"
-#include "MockResultIndexer.hpp"
-#include "MockSendReport.hpp"
+#include "MockDatabaseFeedManager.hpp"
+#include "MockIndexerConnector.hpp"
+
+/**
+ * @brief PackageScanner fake class
+ */
+template<typename TDatabaseFeedManager = MockDatabaseFeedManager>
```

Comment:
Coding style: Could we create a simple and generic "faker" class that receives the ID as a templetized parameter? So we avoid the creation of many classes that are almost identical.

This example illustrates de main idea
<details><summary>Details</summary>
<p>

```c++

#include <iostream>
#include <vector>

constexpr int SOME_CLASS_ID {1};
constexpr int ANOTHER_CLASS_ID {2};
constexpr int OTHER_CLASS_ID {3};

enum class OrchestrationType
{
    Ascending = 0,
    Descending = 1
};

template<int param>
class genericFakeClass
{
private:
    int m_id {param};

public:
    void handleRequest(std::vector<int>& data)
    {
        data.push_back(m_id);
    }
};

template<typename Class1, typename Class2, typename Class3>
class Factory
{
public:
    void static create(OrchestrationType type, std::vector<int>& idVector)
    {
        Class1 class1;
        Class2 class2;
        Class3 class3;
        switch (type)
        {
            case OrchestrationType::Ascending:
                class1.handleRequest(idVector);
                class2.handleRequest(idVector);
                class3.handleRequest(idVector);
                break;
            case OrchestrationType::Descending:
                class3.handleRequest(idVector);
                class2.handleRequest(idVector);
                class1.handleRequest(idVector);
                break;
            default: break;
        }
    }
};

int main()
{
    std::cout << "Hello World\n";

    std::vector<int> idVector;

    Factory<genericFakeClass<SOME_CLASS_ID>, genericFakeClass<ANOTHER_CLASS_ID>, genericFakeClass<OTHER_CLASS_ID>>::
        create(OrchestrationType::Descending, idVector);

    for (auto it = idVector.begin(); it != idVector.end(); ++it)
    {
        std::cout << *it << std::endl;
    }

    return 0;
}
```

</p>
</details>

Reply:
That's the trick I didn't come with, thank you, thank you!!!

Reply:
Great!